app/src/main/java/swati4star/createpdf/util/FileUtils.java

```diff
@@ -378,4 +386,268 @@ public static int checkRepeat(String finalOutputFile, final ArrayList<File> mFil
+                }
+            }else if(isFileUri(uri)) {
+                ret = uri.getPath();
+            }else if(isDocumentUri(ctx, uri)){
+
+                // Get uri related document id.
+                String documentId = DocumentsContract.getDocumentId(uri);
+
+                // Get uri authority.
+                String uriAuthority = uri.getAuthority();
+
+                if(isMediaDoc(uriAuthority))
+                {
+                    String idArr[] = documentId.split(":");
+                    if(idArr.length == 2)
+                    {
+                        // First item is document type.
+                        String docType = idArr[0];
+
+                        // Second item is document real id.
+                        String realDocId = idArr[1];
+
+                        // Get content uri by document type.
+                        Uri mediaContentUri = MediaStore.Images.Media.EXTERNAL_CONTENT_URI;
+                        if("image".equals(docType))
+                        {
+                            mediaContentUri = MediaStore.Images.Media.EXTERNAL_CONTENT_URI;
+                        }else if("video".equals(docType))
+                        {
+                            mediaContentUri = MediaStore.Video.Media.EXTERNAL_CONTENT_URI;
+                        }else if("audio".equals(docType))
+                        {
+                            mediaContentUri = MediaStore.Audio.Media.EXTERNAL_CONTENT_URI;
+                        }
+
+                        // Get where clause with real document id.
+                        String whereClause = MediaStore.Images.Media._ID + " = " + realDocId;
+
+                        ret = getImageRealPath(mContentResolver, mediaContentUri, whereClause);
+                    }
+
+                }else if(isDownloadDoc(uriAuthority))
+                {
+                    // Build download uri.
+                    Uri downloadUri = Uri.parse("content://downloads/public_downloads");
+
+                    // Append download document id at uri end.
+                    Uri downloadUriAppendId = ContentUris.withAppendedId(downloadUri, Long.valueOf(documentId));
+
+                    ret = getImageRealPath(mContentResolver, downloadUriAppendId, null);
+
+                }else if(isExternalStoreDoc(uriAuthority))
+                {
+                    String idArr[] = documentId.split(":");
+                    if(idArr.length == 2)
+                    {
+                        String type = idArr[0];
+                        String realDocId = idArr[1];
+
+                        if("primary".equalsIgnoreCase(type))
+                        {
+                            ret = Environment.getExternalStorageDirectory() + "/" + realDocId;
+                        }
+                    }
+                }
+            }
+        }
+
+        return ret;
+    }
+
+    /* Check whether current android os version is bigger than kitkat or not. */
```

Comment:
The comments format is not consistent in this file. USe javadoc type comments only.